src/cdk/table/table.ts

```diff
@@ -452,7 +481,9 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
       // Optional for backwards compatibility, but a view repeater strategy will always
       // be provided.
       @Optional() @Inject(_VIEW_REPEATER_STRATEGY)
-      protected readonly _viewRepeater: _ViewRepeater<T, RenderRow<T>, RowContext<T>>) {
+      protected readonly _viewRepeater: _ViewRepeater<T, RenderRow<T>, RowContext<T>>,
+      // Optional for backwards compatibility, but a view ruler will always be provided.
+      private readonly _viewportRuler: ViewportRuler) {
```

Comment:
Adding non-optional constructor params here would be a breaking change since we expect that people extend `CdkTable`. Can we make this work with these new params being `@Optional`?

Reply:
Done